### PR TITLE
Ft get an article 59

### PR DESCRIPTION
**Pull request by beblicarl**
#### Pre-Submission Checklist

-   [x] Your pull request targets the `main` branch.
-   [x] Branch name is descriptive and follows the guidelines [here](https://github.com/andela/bestpractices/wiki/Git-naming-conventions-and-best-practices).
-   [x] All tests pass . Use `git commit --amend` to amend any fixes.

#### Checklist:

-   [x] All tests are passing.
-   [x] Tested changes locally.
-   [x] Improved documentation or readme where necessary.
-   [x] Your commit messages format follows our commit message guideline [here](https://github.com/andela/bestpractices/wiki/Git-naming-conventions-and-best-practices#commits).

#### What does this PR do?

It gets you an article by id
#### How should this be manually tested?

1. Start the server with` npm start ` 
2. Make a get request to this endpoint http://localhost:3000/api/v1/articles/<:postId> 
3. This will successfully return the article with the particular URL param

##### Any background context you want to provide?

It allows an employee to search for an article by id
#### What are the relevant issues?
This endpoint returns articles by their id

#### Screenshots (if appropriate)
image

#### Does it require any deployments updates ?

https://github.com/tbf-challenge/teamwork-be/compare/ft-get-an-article-59?expand=1

### CODE REVIEW

**Comment by beblicarl** [comment]
@jocrah , kindly review this for me

File: src/routes/posts.js [hunk 1]
```diff
@@ -33,6 +34,42 @@ const createPost = async (req, res, next) => {
...
+// GET POST BY ID
+
+const getPost = async (req, res, next) => {
+    const { id } = req.params
+    try {
+        const posts = await db.query(
+            'SELECT p.*, jsonb_agg(c.* ORDER BY c."createdAt" DESC) as comments FROM posts p LEFT JOIN comments c ON p.id = c."postId" WHERE p.id=$1 GROUP BY p.id;',
+            [id]
+        )
+        const article = posts.rows[0]
```
[the remark this answers is not in the document]
Comment (beblicarl) [reply src/routes/posts.js:48]: alright

[the remark this answers is not in the document]
Comment (beblicarl) [reply src/routes/posts.js:48]: I have made the changes now @jocrah

[the remark this answers is not in the document]
Comment (jocrah) [reply src/routes/posts.js:48]: doesn't seem like you've pushed it

[the remark this answers is not in the document]
Comment (beblicarl) [reply src/routes/posts.js:48]: > doesn't seem like you've pushed it

I can confirm it has been pushed now

File: src/routes/posts.js [hunk 2]
```diff
@@ -33,6 +34,49 @@ const createPost = async (req, res, next) => {
...
 
+// GET REQUESTS
+
+// GET POST BY ID
+
+const getPost = async (req, res, next) => {
+    const { id } = req.params
+    try {
+        const posts = await db.query(
+            'SELECT p.*, jsonb_agg(c.* ORDER BY c."createdAt" DESC) as comments FROM posts p LEFT JOIN comments c ON p.id = c."postId" WHERE p.id=$1 GROUP BY p.id;',
```
Comment (jocrah) [inline src/routes/posts.js:45]: @beblicarl  can we use a backtick(`) to do a multiline query here?

something like

```
`
SELECT p.*, jsonb_agg(c.* ORDER BY c."createdAt" DESC) as comments
FROM posts p 
LEFT JOIN comments c ON p.id = c."postId"
WHERE p.id=$1
GROUP BY p.id;

`
```

this is to make it a bit more readable by making the keywords start each line

File: src/routes/posts.js [hunk 3]
```diff
@@ -33,6 +34,49 @@ const createPost = async (req, res, next) => {
...
+            [id]
+        )
+        const article = posts.rows[0]
+
+        if (!article) {
+            res.status(404).json({
```
Comment (jocrah) [inline src/routes/posts.js:51]: just me nitpicking. we can use a `return` here so we don't have to use an `else` on line `55` later

File: src/routes/posts.js (hunk 2 above)
Comment (beblicarl) [reply src/routes/posts.js:45]: alright

File: src/routes/posts.js (hunk 3 above)
Comment (beblicarl) [reply src/routes/posts.js]: i have made the changes and pushed to the repo

**Review by jocrah — APPROVED** [review]

**Pull request merged**